src/pages/Facility/settings/devices/components/ServiceHistoryForm.tsx

```diff
@@ -0,0 +1,197 @@
+import { zodResolver } from "@hookform/resolvers/zod";
+import { useMutation } from "@tanstack/react-query";
+import { format } from "date-fns";
+import { CalendarIcon } from "lucide-react";
+import { useEffect } from "react";
+import { useForm } from "react-hook-form";
+import { useTranslation } from "react-i18next";
+import { toast } from "sonner";
+import { z } from "zod";
+
+import { cn } from "@/lib/utils";
+
+import { Button } from "@/components/ui/button";
+import { Calendar } from "@/components/ui/calendar";
+import {
+  Form,
+  FormControl,
+  FormField,
+  FormItem,
+  FormLabel,
+  FormMessage,
+} from "@/components/ui/form";
+import {
+  Popover,
+  PopoverContent,
+  PopoverTrigger,
+} from "@/components/ui/popover";
+import { Textarea } from "@/components/ui/textarea";
+
+import mutate from "@/Utils/request/mutate";
+import { ServiceHistory } from "@/types/device/device";
+import deviceApi from "@/types/device/deviceApi";
+
+interface Props {
+  facilityId: string;
+  deviceId: string;
+  serviceRecord?: ServiceHistory | null;
+  onSubmitSuccess: (serviceRecord: ServiceHistory) => void;
+}
+
+export default function ServiceHistoryForm({
+  facilityId,
+  deviceId,
+  serviceRecord,
+  onSubmitSuccess,
+}: Props) {
+  const { t } = useTranslation();
+  const isEditMode = !!serviceRecord;
+
+  const formSchema = z.object({
+    note: z.string().min(1, { message: t("field_required") }),
+    serviced_on: z
+      .date({ required_error: t("field_required") })
+      .min(new Date(new Date().setHours(0, 0, 0, 0)), {
+        message: t("service_date_min_date"),
+      }),
```

Comment:
this should be totally the opposite right? 

max should be today right?

Reply:
🤔  Yeah makes sense.

Added the min since earlier req was to `restrict the submission of service history to a future date`